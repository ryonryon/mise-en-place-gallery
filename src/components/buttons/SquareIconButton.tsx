import styled from "@emotion/styled";
import * as React from "react";

interface Props extends React.ButtonHTMLAttributes<HTMLButtonElement> {
  height: string;
  width: string;
  value: string;
  children: React.ReactNode;
  disabled: boolean;
}

const SquareIconButton = ({ height, width, value, children, disabled, ...rest }: Props) => {
  const StyledButton = styled.button`
    height: ${height};
    width: ${width};
    background-color: rgba(225, 225, 225, 0.3);
    border: none;
    border-radius: 0;
    padding: 7.5px;

    &:hover {
      background-color: rgba(225, 225, 225, ${disabled ? "0.3" : "0.9"});
      transition: all 0.5s ease-out;
    }
  `;

  return (
    <StyledButton value={value} disabled={disabled} {...rest}>
      {children}
    </StyledButton>
  );
};

SquareIconButton.defaultProps = {
  height: "45px",
  width: "45px",
  value: "squareIconButton",
  children: null,
  disabled: false
};

export default SquareIconButton;
