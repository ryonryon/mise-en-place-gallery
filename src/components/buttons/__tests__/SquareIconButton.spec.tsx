import * as React from "react";
import * as renderer from "react-test-renderer";

import SquareIconButton from "../SquareIconButton";

describe("SquareIconButton test", () => {
  it("tests SquareIconButton renders correctly ", () => {
    const squareIconButton = renderer.create(<SquareIconButton />).toJSON();
    expect(squareIconButton).toMatchSnapshot();
  });
});
