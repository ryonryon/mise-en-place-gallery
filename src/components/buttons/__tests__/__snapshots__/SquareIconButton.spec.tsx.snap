// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SquareIconButton test tests SquareIconButton renders correctly  1`] = `
<button
  className="css-1o5go31"
  disabled={false}
  value="squareIconButton"
/>
`;
